src/Jenssegers/Mongodb/Eloquent/Model.php

```diff
@@ -99,7 +99,7 @@ protected function asDateTime($value)
     {
         // Convert UTCDateTime instances.
         if ($value instanceof UTCDateTime) {
-            return Carbon::createFromTimestamp($value->toDateTime()->getTimestamp());
+            return Carbon::createFromTimestampMs($value->toDateTime()->format('Uv'));
```

Comment:
```suggestion
            return Date::createFromTimestampMs($value->toDateTime()->format('Uv'));
```

Reply:
What is the profit to use `Date` instead of `Carbon`?

Reply:
Sorry, i see this comment https://github.com/jenssegers/laravel-mongodb/pull/1963#pullrequestreview-359367649